buildSrc/src/main/kotlin/com/playframework/gradle/DocumentationTestPlugin.kt

```diff
@@ -19,18 +19,22 @@ class DocumentationTestPlugin : Plugin<Project> {
                 groovy.srcDir("src/docTest/groovy")
             }
             resources.srcDir("src/docTest/resources")
-            compileClasspath += sourceSets["main"]!!.output + sourceSets["integTestFixtures"]!!.output + testRuntimeClasspath + integTestFixturesRuntimeClasspath
+            val main by sourceSets
+            val integTestFixtures by sourceSets
+            compileClasspath += main.output + integTestFixtures.output + testRuntimeClasspath + integTestFixturesRuntimeClasspath
             runtimeClasspath += output + compileClasspath
         }
 
-        val docTest by tasks.creating(Test::class) {
+        val docTest by tasks.registering(Test::class) {
```

Comment:
I have a question about this. In case the plugin is registered with `register` is there a benefit to also "registering" the tasks with `register`? Or is there no correlation?

Reply:
Plugins aren't registered/lazy. What do you mean?

Reply:
I mean this specifically: https://github.com/gradle/play/blob/master/buildSrc/build.gradle.kts#L21. It uses the same terminology. It's impossible to understand for a user to grasp the underlying implementation details if the terms is the same as for registering tasks but the behavior is different.

Reply:
Plugins in the `gradlePlugin.plugins` container context, got it.

Registering on a container always means the same thing: creating and configuration will happen later, when the element is eagerly requested.

I don't think there's a connection between `gradlePlugin.plugins` and the `docTest` task registration above though.